crits/emails/handlers.py

```diff
@@ -560,12 +553,10 @@ def handle_email_fields(data, analyst, method, related_id=None, related_type=Non
             retVal['message'] = 'Related Object not found.'
             return retVal
 
-    if related_obj:
-        relationship_type=RelationshipTypes.inverse(relationship=relationship_type)
-        new_email.add_relationship(related_obj,
-                                          relationship_type,
-                                          analyst=analyst,
-                                          get_rels=False)
+
+    new_email.add_source(source=sourcename, method=method,
+                         reference=reference, analyst=analyst)
+
 
     try:
         new_email.save(username=analyst)
```

Comment:
The email gets saved here after the Source and Campaign get added.  Relationships aren't formed until later.